testing/integration/package_version_test.go

```diff
@@ -93,20 +90,31 @@ func TestComponentBuildHashInDiagnostics(t *testing.T) {
 		"failed to install start agent [output: %s]", string(output))
 
 	stateBuff := bytes.Buffer{}
-	isHealth := func() bool {
+	allHealthy := func() bool {
 		stateBuff.Reset()
 
-		err := f.IsHealthy(ctx)
+		status, err := f.ExecStatus(ctx)
 		if err != nil {
-			stateBuff.WriteString(err.Error())
+			stateBuff.WriteString(fmt.Sprintf("failed to get agent status: %v",
+				err))
 			return false
 		}
 
-		return true
+		allHealthy := true
+		for _, c := range status.Components {
+			state := client.State(c.State)
+			if state != client.Healthy {
+				allHealthy = false
+				stateBuff.WriteString(fmt.Sprintf("%s not health, state: %s",
```

Comment:
Nitpick: couldn't you just track the last status and dump that if the assertion fails? I understand that it would be bigger but it's a test and if it fails, the more information we have the better, I think